Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -25,6 +25,15 @@ public IonObjectSerializer(IonSerializer ionSerializer, IonSerializationOptions
 
         public override object Deserialize(IIonReader reader)
         {
+            var customObjectSerializer = this.GetCustomObjectSerializer();
```

Comment:
I feel like this becomes simpler by using the TryCustomObjectDeserialize paradigm instead.

Reply:
Done.